### DISCUSSION TITLE
Some question about `stage`

### DISCUSSION THREAD

**Discussion by sologgfun** [kubernetes-sigs/kwok#300] [General]
Im trying to help complete the pr [[kwok] Simulate the services provided by Kubelet #162](https://github.com/kubernetes-sigs/kwok/issues/162)
But I encountered some problems 😭
Im my mind , `stage` is a crd use to manage node or pod lifecycle.
I follow the docs create a cluster `https://kwok.sigs.k8s.io/docs/contributing/development/`
I notice this describe 
> The node-initialize Stage will be applied to nodes that do not have any conditions set in their status.conditions field. When applied, this Stage will set the status.conditions field for the node, as well as the status.addresses, status.allocatable, and status.capacity fields.

But after the cluster created. There is no crd in the new cluster,And when i put a new node or pod in the cluster, node addresses is filled with
```
addresses:
- type: Hostname
address: kwok-kwok-kwok-controller
```

So , My question is 
1. where can i find crd named `stage`? 
2. why these field `status.addresses, status.allocatable, and status.capacity` be filled without `stage`?

Did I miss something? If any refrence can give will be great , thank you!

**Comment by wzshiming** [comment]
> where can i find crd named stage?

Currently, they are only locally configured with `--config` passed in, and will support CRD  #264 for configuration in the future when the API is more stable. 

> why these field status.addresses, status.allocatable, and status.capacity be filled without stage?

https://github.com/kubernetes-sigs/kwok/blob/cef98dea6bd74b1fcd5262f601bc293196c72fa5/stages/node-fast.yaml#L47

**Reply by sologgfun** [reply]
fine, thank you again!😄
